File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:

        vowels=set('aeiouAEIOU')
        s=list(s)
        left=0
        right=len(s)-1

        while right>left:
            if s[left] in vowels and s[right] in vowels:
                s[left],s[right]=s[right],s[left]
                left=left+1
                right=right-1
            elif s[right] in vowels:
                left=left+1
            elif s[left] in vowels:
                right=right-1
            else:
                left=left+1
                right=right-1
            
        return '' .join(s)














        # arr=list(s)
        # new_array=[]
        # n=len(arr)
        # for i in range(n):
        #   if arr[i]=="a" or arr[i]=="e" or arr[i]=="i"or arr[i]=="o" or arr[i]=="u" or arr[i]=="A" or arr[i]=="E" or arr[i]=="I"or arr[i]=="O" or arr[i]=="U":
        #         new_array.append(arr[i])

        # new_array.reverse()
        # vowel_count=0
        
        # for i in range(n):
        #     if arr[i]=="a" or arr[i]=="e" or arr[i]=="i"or arr[i]=="o" or arr[i]=="u" or arr[i]=="A" or arr[i]=="E" or arr[i]=="I"or arr[i]=="O" or arr[i]=="U":
        #         arr[i]=new_array[vowel_count % len(new_array)]
        #         vowel_count=vowel_count+1
                
        # out= ''.join(arr)
        # return out


        

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {

        char [] charArray=s.toCharArray();

      


        String vowels="aeiouAEIOU";

        int start=0;
        int end=s.length()-1;

        
        while (start<end){
            while(start<end && vowels.indexOf(charArray[start])==-1){
                start++;
            }
            while (start<end && vowels.indexOf(charArray[end])==-1){
                end--;
            }

            char temp=charArray[start];
            charArray[start]=charArray[end];
            charArray[end]=temp;

            start++;
            end--;
         
        }
        return new String(charArray);




        
    }
}